### PR TITLE
kaput-cli: init at 1.1.1

### PR DESCRIPTION
**Pull request by vrinek**
###### Motivation for this change

`kaput-cli` is a CLI for put.io. It helps me script some of my workflow and I think others might find it useful as well.

This is my first time adding a package, so please point out anything I have missed.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
```
❯ nix path-info --json --all | jq -r 'sort_by(.registrationTime)[-11:-1][].path' | grep kaput | xargs nix path-info -Shs
/nix/store/fqia559rfp7ziplndpyzfjq1vzzimggh-kaput-cli-modules-1.1.1       43.1M  254.1M
/nix/store/p80bq37vg2dc2vwr9i86f48fxyna4n87-kaput-cli                     43.3M  297.4M
```
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -10186,6 +10186,12 @@
     email = "vq@erq.se";
     name = "Daniel Nilsson";
   };
+  vrinek = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:10189]: Please put this into a separate commit with the message ``maintainers: add vrinek``.

File: pkgs/tools/networking/kaput-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{ lib, mkYarnPackage, fetchFromGitHub }:
+
+mkYarnPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/kaput-cli/default.nix:3]: This should be probably converted to be in nodePackages.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (vrinek) [reply maintainers/maintainer-list.nix:10189]: Resolved in https://github.com/NixOS/nixpkgs/pull/117110/commits/c26a559134f8dc524e8cc057e2cd427143ed104e.

File: pkgs/tools/networking/kaput-cli/default.nix (hunk 2 above)
Comment (vrinek) [reply pkgs/tools/networking/kaput-cli/default.nix:3]: I am having a hard time tracking down some sort of guide to get this done. I checked https://nixos.org/manual/nixpkgs/stable/#node.js which pointed me to https://github.com/svanderburg/node2nix, but I cannot find any explicit information on how to prepare a `default.nix` for nixpkgs.

I've also tried reading through some existing packages that make use of `nodePackages` but could not figure things out.

Do you have anything to suggest?

Comment (vrinek) [reply pkgs/tools/networking/kaput-cli/default.nix]: I've added the npm package to `pkgs/development/node-packages/node-packages.json`. I hope this was what you meant initially.

It builds and works find as `nodePackages.kaput-cli` but it does not build on latest (`nodePackages_latest.kaput-cli`).

Is this ok to leave as is, or is there a way to mark the broken one as such?

Comment (SuperSandro2000) [reply pkgs/tools/networking/kaput-cli/default.nix]: > I've added the npm package to pkgs/development/node-packages/node-packages.json. I hope this was what you meant initially.

Yes

> It builds and works find as nodePackages.kaput-cli but it does not build on latest (nodePackages_latest.kaput-cli).

The default is nodePackages and I usually don't worry about nodePackages_latest.
